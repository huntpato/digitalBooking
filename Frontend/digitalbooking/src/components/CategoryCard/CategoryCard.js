import React from "react";
import { useFetch } from "use-http";
import styles from "./CategoryCard.module.css";

export const CategoryCard = (props) => {
  
  const options = {headers:{
    "X-Requested-With": "XMLHttpRequest"
  }};
  const { categoryCards } = styles;
  const { image, title, /*description,*/ alt } = props;
  const { data = [] } = useFetch("https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/category/count/Hoteles" , options, []);

  return (
    <article className={categoryCards}>
      <img src={image} alt={alt} />
      <h5>{title}</h5>
      <p>{data} {title}</p>
    </article>
  );
};
