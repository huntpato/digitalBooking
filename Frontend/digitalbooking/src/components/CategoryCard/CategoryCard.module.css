
.categoryCards {
    border-radius: 5px;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
    width: 100%;
    height: 280px;
}

.categoryCards:hover{
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.23);
}


.categoryCards h5{
    font-size: 15px;
    margin:6px 0px 3px 10px ;
}

.categoryCards p{
    font-size: 11px;
    margin:0px 0px 0px 10px ;
}

.categoryCards img{
    border-radius: 5px 5px 0px 0px;
    height: 78%;
    width: 100%;
    object-fit:cover;
}


@media (max-width: 768px){

    .categoryCards {
        height: 270px;
    }

}

@media (max-width: 576px){

    .categoryCards {
        height: 390px;
    }

    .categoryCards img{
        height: 84%;    
    }   

}


@media (max-width: 480px){

    .categoryCards {
        height: 350px;
    }

    .categoryCards img{
         height: 82%;
    }

}


