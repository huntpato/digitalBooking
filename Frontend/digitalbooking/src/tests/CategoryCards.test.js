import React from 'react';
import { shallow } from 'enzyme';
import { CategoryCard } from '../components/CategoryCard/CategoryCard'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Comprobar el correcto funcionamiento del componente <CategoryCard/>', () => {

    const image= "https://images.unsplash.com/photo-1584704876450-0eb3fd5a2f65?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=774&q=80";
    const title= "Hermoso departamento en la playa."
    const desc = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt.";
    const alt = "Lorem ipsum";
    const wrapper= shallow(<CategoryCard image={image} title={title} description={desc} alt={alt} />);
    
    test('Verificar que el componente renderiza corectamente ', () => {
        
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const card = wrapper.find('article');
        expect(card.hasClass('categoryCards')).toBe(true);
    })
    
})
