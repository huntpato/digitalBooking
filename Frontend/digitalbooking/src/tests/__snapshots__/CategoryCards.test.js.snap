// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comprobar el correcto funcionamiento del componente <CategoryCard/> Verificar que el componente renderiza corectamente  1`] = `
<article
  className="categoryCards"
>
  <img
    alt="Lorem ipsum"
    src="https://images.unsplash.com/photo-1584704876450-0eb3fd5a2f65?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=774&q=80"
  />
  <h5>
    Hermoso departamento en la playa.
  </h5>
  <p>
     
    Hermoso departamento en la playa.
  </p>
</article>
`;
